KMD.Identity.TestApplications.OpenID.Angular/ClientApp/src/app/user-delegation/user-delegation.service.ts

```diff
@@ -11,12 +12,25 @@ export class UserDelegationService {
   private delegationToken = new BehaviorSubject<any>(null);
   public delgationToken$ = this.delegationToken.asObservable();
 
-  constructor(private  oidcSecurityService: OidcSecurityService) { 
+  constructor(
+    private authenticationContext: AuthenticationContext,
+    private oidcSecurityService: OidcSecurityService) {
+
+      this.authenticationContext.tokenExchangeLogin$.subscribe(loginResponse => {
+        const token = this.oidcSecurityService.getUserData(ConfigIds.UserDelegation);
+        this.delegationToken.next(token);
+      })
   }
 
-  public requestDelegationToken() {
-    // maybe this isn't needed to be observable at all.
-    let userData = this.oidcSecurityService.getUserData(ConfigIds.Code)
-    this.delegationToken.next(userData);
+  public requestTokenExchange() {
+
+    const authOption: AuthOptions = {
+      customParams: {
+      "userdelegation_actas_nameid": "myself-for-now-until-implementation-is-provided",
```

Comment:
This is a fair representation of my thoughts when authenticating requesting a user delegation